src/PlayFabBuddy.Lib/Commands/Player/DeletePlayersBySegmentCommand.cs

```diff
@@ -0,0 +1,64 @@
+﻿using PlayFabBuddy.Lib.Adapter.Accounts;
+using PlayFabBuddy.Lib.Interfaces.Adapter;
+
+namespace PlayFabBuddy.Lib.Commands.Player;
+
+/// <summary>
+/// Command to delete players by segment
+/// </summary>
+public class DeletePlayersBySegmentCommand
+{
+    private readonly IPlayStreamAdapter playStreamAdapter;
+    private readonly IPlayerAccountAdapter playerAccountAdapter;
+
+    public DeletePlayersBySegmentCommand(IPlayStreamAdapter playStreamAdapter, IPlayerAccountAdapter playerAccountAdapter)
+    {
+        this.playStreamAdapter = playStreamAdapter;
+        this.playerAccountAdapter = playerAccountAdapter;
+    }
+
+    /// <summary>
+    /// Deletes all Master Player Accounts from a given segment, identified by <paramref name="segmentName"/>.
+    /// Reports back progress via <paramref name="progress"/>
+    /// </summary>
+    /// <param name="segmentName">The segment name from which Master Player Accounts shall be deleted from</param>
+    /// <param name="progress">A progress object to report progress on</param>
+    /// <returns>The number of deleted Master Player Accounts</returns>
+    public async Task<(int removedCount, List<MasterPlayerAccountAdapter> playersInSegment)> ExecuteAsync(string segmentName, IProgress<double> progress)
+    {
+        var allPlayersSegmentId = await this.playStreamAdapter.GetSegmentById(segmentName);
```

Comment:
are we expecting the segment id for all players here? If not, we should find a better name like `segmentId`

Reply:
fixed, was a leftover from an early implementation